packages/xo-server/src/api/vm.mjs

```diff
@@ -1327,6 +1329,97 @@ importFromEsxi.params = {
   vm: { type: 'string' },
 }
 
+/**
+ * on success:  returns an object, the keys are the esxi id, and the values are the created vm uuid
+ * On error: throw an error. If stopOnError is false, continue when an error occurs, throws an error at the end with a 'succeeded'
+ * property listing the VM properly imported and a 'errors' property with all the errors collected
+ */
+export async function importMultipleFromEsxi({
+  concurrency = 2,
+  host,
+  network,
+  password,
+  sr,
+  sslVerify = true,
+  stopSource = false,
+  stopOnError = false,
+  thin = false,
+  user,
+  vms,
+}) {
+  const task = await this.tasks.create({ name: `importing vms ${vms.join(',')}` })
+  let done = 0
+  return task.run(async () => {
+    Task.set('total', vms.length)
+    Task.set('done', 0)
+    Task.set('progress', 0)
+    const result = {}
+    try {
+      await asyncEach(
+        vms,
+        async vm => {
+          await new Task({ name: `importing vm ${vm}` }).run(async () => {
+            try {
+              const vmUuid = await this.migrationfromEsxi({
+                host,
+                user,
+                password,
+                sslVerify,
+                thin,
+                vm,
+                sr,
+                network,
+                stopSource,
+              })
+              result[vm] = vmUuid
+            } finally {
+              done++
+              Task.set('done', done)
+              Task.set('progress', Math.round((done * 100) / vms.length))
+            }
+          })
+        },
+        {
+          concurrency,
+          stopOnError,
+        }
+      )
+      return result
+    } catch (error) {
+      // error is an AggregatedError containing all the errors if stopOnError is false
+      // it is the first error otherwise
+
+      // add the successfull VMs if the UX want to show them something positive
```

Comment:
I think we don't need this comment because you explain it a bit in the comment above.